sources/packages/web/src/components/students/ApplicationDetails.vue

```diff
@@ -0,0 +1,232 @@
+<template>
+  <v-container>
+    <v-card>
+      <formio
+        formName="studentapplicationdetails"
+        :data="initialData"
+        @customEvent="customEventCallback"
+      ></formio>
+    </v-card>
+    <v-card>
+      <v-container>
+        <p class="category-header-large">Track your application</p>
+        <div class="progress">
+          <div
+            class="progress-bar bg-secondary"
+            :class="getTrackingClass()"
+            role="progressbar"
+            :style="progressBarStyle"
+            aria-valuemax="100"
+          >
+            {{ progressBarLabel }}
+          </div>
+        </div>
+        <v-row class="mt-1 mb-2">
+          <v-col>{{ ApplicationStatus.submitted }}</v-col>
+          <v-col
+            >{{ ApplicationStatus.inProgress }}
+            <i class="mr-2" :class="inProgressIconClass" aria-hidden="true"></i
+          ></v-col>
+          <v-col>{{ ApplicationStatus.assessment }} </v-col>
+          <v-col
+            >{{ ApplicationStatus.enrollment }}
+            <i class="mr-2" :class="enrollmentIconClass" aria-hidden="true"></i
+          ></v-col>
+          <v-col
+            >{{ ApplicationStatus.completed }}
+            <i class="mr-2" :class="completeIconClass" aria-hidden="true"></i
+          ></v-col>
+        </v-row>
+
+        <formio
+          formName="trackstudentapplication"
+          :data="dataForTracking"
+          @customEvent="customEventCallback"
+        ></formio></v-container
+    ></v-card>
+  </v-container>
+</template>
+<script lang="ts">
+import { useRouter } from "vue-router";
+import { onMounted, ref } from "vue";
+import formio from "@/components/generic/formio.vue";
+import { StudentRoutesConst } from "@/constants/routes/RouteConstants";
+import {
+  FormIOCustomEvent,
+  FormIOCustomEventTypes,
+  ApplicationStatus,
+  GetApplicationDataDto,
+  StudentApplicationDetails,
+  StudentApplicationDetailsForTracking,
+  COEStatus,
+  ProgramInfoStatus,
+} from "@/types";
+
+export default {
+  components: {
+    formio,
+  },
+  props: {
+    applicationDetails: {
+      type: Object,
+      required: true,
+      default: {} as GetApplicationDataDto,
+    },
+  },
+  setup(props: any) {
+    const initialData = ref({} as StudentApplicationDetails);
+    const dataForTracking = ref({} as StudentApplicationDetailsForTracking);
+    const router = useRouter();
+    const progressBarStyle = ref();
+    const progressBarLabel = ref();
+    const inProgressIconClass = ref();
+    const enrollmentIconClass = ref();
+    const completeIconClass = ref();
+    const getTrackingClass = () => {
+      if (
+        ApplicationStatus.draft === props.applicationDetails.applicationStatus
```

Comment:
**Not a blocker, not asking to change it** but, I see that we are using sometimes the Yoda here in there. I saw inside the same PR (not this one), it was used and not used (inside the same code change). I am still struggling to see any benefit to using it in typescript. I would like to make the point again that it, IMO, makes the code harder to read, and the benefit of using it is almost inexistent.

![image](https://user-images.githubusercontent.com/61259237/134995728-b35739f2-c7cf-445a-a3d6-75fcc41bbdad.png)
Source: https://knowthecode.io/yoda-conditions-yoda-not-yoda

@guru-aot and @dheepak-aot FYI